stacks-signer/src/runloop.rs

```diff
@@ -13,1050 +13,288 @@
-                .unwrap_or_else(|e| {
-                    warn!("Failed to submit block for validation: {:?}", e);
-                });
-        }
-    }
-
-    /// Process inbound packets as both a signer and a coordinator
-    /// Will send outbound packets and operation results as appropriate
-    fn handle_packets(&mut self, res: Sender<Vec<OperationResult>>, packets: &[Packet]) {
-        let signer_outbound_messages = self
-            .signing_round
-            .process_inbound_messages(packets)
-            .unwrap_or_else(|e| {
-                error!("Failed to process inbound messages as a signer: {e}");
-                vec![]
-            });
-
-        // Next process the message as the coordinator
-        let (coordinator_outbound_messages, operation_results) = self
-            .coordinator
-            .process_inbound_messages(packets)
-            .unwrap_or_else(|e| {
-                error!("Failed to process inbound messages as a coordinator: {e}");
-                (vec![], vec![])
-            });
-
-        if !operation_results.is_empty() {
-            // We have finished a signing or DKG round, either successfully or due to error.
-            // Regardless of the why, update our state to Idle as we should not expect the operation to continue.
-            self.state = State::Idle;
-            self.process_operation_results(&operation_results);
-            self.send_operation_results(res, operation_results);
-        }
-        self.send_outbound_messages(signer_outbound_messages);
-        self.send_outbound_messages(coordinator_outbound_messages);
-    }
+            return Ok(None);
+        };
 
-    /// Validate a signature share request, updating its message where appropriate.
-    /// If the request is for a block it has already agreed to sign, it will overwrite the message with the agreed upon value
-    /// Returns whether the request is valid or not.
-    fn validate_signature_share_request(&self, request: &mut SignatureShareRequest) -> bool {
-        let message_len = request.message.len();
-        // Note that the message must always be either 32 bytes (the block hash) or 33 bytes (block hash + b'n')
-        let hash_bytes = if message_len == 33 && request.message[32] == b'n' {
-            // Pop off the 'n' byte from the block hash
-            &request.message[..32]
-        } else if message_len == 32 {
-            // This is the block hash
-            &request.message
-        } else {
-            // We will only sign across block hashes or block hashes + b'n' byte
-            debug!("Signer #{}: Received a signature share request for an unknown message stream. Reject it.", self.signer_id);
-            return false;
+        // We can only register for a reward cycle if a reward set exists. We know that it should exist due to our earlier check for reward_set_calculated
+        let Some(reward_set_signers) = self.stacks_client.get_reward_set(reward_cycle)?.signers
```

Comment:
By slot id do you mean page id? cause what I am calculating is the 0 vs 1 that the signer needs to determine its stacker db contract to write to.

Reply:
Oh good you pointed this out because the make_Stackerdb_contract function call changed from taking the reward cycle % 2 to taking the reward cycle and doing it locally. I missed this in a merge

Reply:
I know I actually tried this earlier. The signer Id as far as I am aware is the offset in the reward set entry that gets returned. THe signer slot id is not this however. This has to be retreieved from the contract itself. I have verified that they appear to be different...
```
    /// The index into the signers list of this signer's key (may be different from signer_id)
    pub signer_slot_id: u32,
    /// The signer ID assigned to this signer
    pub signer_id: u32,
 ```